examples/linear_model/plot_ard.py

```diff
@@ -1,119 +1,155 @@
 """
-==================================================
-Automatic Relevance Determination Regression (ARD)
-==================================================
+==========================================
+Comparison of Linear Models for Regression
+==========================================
 
-Fit regression model with Bayesian Ridge Regression.
+This example compares three kind of linear models for regression:
 
-See :ref:`bayesian_ridge_regression` for more information on the regressor.
+ - a :ref:`ordinary_least_squares`
+ - a :ref:`automatic_relevance_determination`
+ - a :ref:`bayesian_ridge_regression`
 
-Compared to the OLS (ordinary least squares) estimator, the coefficient
-weights are slightly shifted toward zeros, which stabilises them.
+The comparisons are based on the models' coefficients with respect to the true
+coefficients.
 
-The histogram of the estimated weights is very peaked, as a sparsity-inducing
-prior is implied on the weights.
-
-The estimation of the model is done by iteratively maximizing the
+The estimation of the model is done in both cases by iteratively maximizing the
 marginal log-likelihood of the observations.
 
-We also plot predictions and uncertainties for ARD
-for one dimensional regression using polynomial feature expansion.
-Note the uncertainty starts going up on the right side of the plot.
-This is because these test samples are outside of the range of the training
-samples.
+We also plot predictions and uncertainties for the ARD and the Bayesian Ridge
+regressions using a polynomial feature expansion to fit a non-linear feature.
+Notice that the ARD regression captures the ground truth the best, but due to
+the limitations of a polynomial regression, both models fail when extrapolating.
 
 """
 
-import numpy as np
-import matplotlib.pyplot as plt
-from scipy import stats
-
-from sklearn.linear_model import ARDRegression, LinearRegression
-
 # %%
-# Generating simulated data with Gaussian weights
-
-# Parameters of the example
-np.random.seed(0)
-n_samples, n_features = 100, 100
-# Create Gaussian data
-X = np.random.randn(n_samples, n_features)
-# Create weights with a precision lambda_ of 4.
-lambda_ = 4.0
-w = np.zeros(n_features)
-# Only keep 10 weights of interest
-relevant_features = np.random.randint(0, n_features, 10)
-for i in relevant_features:
-    w[i] = stats.norm.rvs(loc=0, scale=1.0 / np.sqrt(lambda_))
-# Create noise with a precision alpha of 50.
-alpha_ = 50.0
-noise = stats.norm.rvs(loc=0, scale=1.0 / np.sqrt(alpha_), size=n_samples)
-# Create the target
-y = np.dot(X, w) + noise
+# Generate synthetic dataset
+# --------------------------
+from sklearn.datasets import make_regression
+
+X, y, w = make_regression(
+    n_samples=100,
+    n_features=100,
```

Comment:
It should be worth adding an additional note that here we are in a system where `n_samples == n_features`. This setting is challenging for an OLS and having a prior on the weights avoids having arbitrary large weights